test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -492,6 +500,88 @@ add_specs suite_builder =
+            huge_b.to_integer . should_equal huge_integer
+            huge_c.to_integer . should_equal huge_integer
+
+            (-a).to_integer . should_equal -12000
+            (-b).to_integer . should_equal -12000
+            (-c).to_integer . should_equal -12000
+            (-huge_a).to_integer . should_equal -huge_integer
+            (-huge_b).to_integer . should_equal -huge_integer
+            (-huge_c).to_integer . should_equal -huge_integer
+
+        group_builder.specify "should convert correctly to Float" <|
+            a = Decimal.new "12000"
+            b = Decimal.new "12000" (Math_Context.new 2)
+            c = Decimal.new "12000" (Math_Context.new 3)
+            huge_a = Decimal.new "3.4E300"
+            huge_b = Decimal.new "3.400E300"
+            huge_c = Decimal.new "3.4000000E300"
+            huge_float = 3.4 * (10 ^ 300)
+
+            a.to_float . should_equal 12000.0
+            b.to_float . should_equal 12000.0
+            c.to_float . should_equal 12000.0
+            huge_a.to_float . should_equal huge_float
+            huge_b.to_float . should_equal huge_float
+            huge_c.to_float . should_equal huge_float
+
+            (-a).to_float . should_equal -12000
+            (-b).to_float . should_equal -12000
+            (-c).to_float . should_equal -12000
+            (-huge_a).to_float . should_equal -huge_float
+            (-huge_b).to_float . should_equal -huge_float
+            (-huge_c).to_float . should_equal -huge_float
+
+        group_builder.specify "attaches a warning to the result of .to_float when it is out the representable range" <|
+            huge_a = Decimal.new "3.4E320"
+            f = huge_a.to_float
+            f.should_equal Number.positive_infinity
+            Problems.expect_only_warning Out_Of_Range f
+            f2 = (-huge_a).to_float
+            f2.should_equal Number.negative_infinity
+            Problems.expect_only_warning Out_Of_Range f2
+
+        group_builder.specify "attaches a warning to the result of .to_float when it results in a loss of precision" <|
+            Problems.expect_only_warning Loss_Of_Numeric_Precision (Decimal.new "0.1" . to_float)
+
+        group_builder.specify "should convert correctly to Integer and Float using implicit conversions" <|
+            i = Integer.from (Decimal.new "5634")
+            i.should_be_a Integer
+            i.should_equal 5634
+
+            f = Float.from (Decimal.new "56.34")
+            f.should_be_a Float
+            f.should_equal 56.34
+
+        group_builder.specify "Decimal.to_float should compare correctly with the original Decimal" <|
+            huge_a = Decimal.new "3.4E300"
+            huge_a_float = Float.from huge_a
+
+            (huge_a_float == huge_a) . should_be_true
+            (huge_a == huge_a_float) . should_be_true
+
+        group_builder.specify "Decimal.to_integer should compare correctly with the original Decimal" pending="Decimal.to_integer comparison" <|
```

Comment:
I cannot run this test as:
```
enso$ ./built-distribution/enso-engine-*/enso-*/bin/enso --run test/Base_Tests/ "Decimal.to_integer"
```
probably because `Decimal_Spec` isn't listed in `Main.enso`. Is it even executed on the CI? I don't see [Decimal.to_integer](https://github.com/enso-org/enso/actions/runs/8347396744/job/22846918263?pr=9462) anywhere in the log output!

Reply:
It is listed in https://github.com/enso-org/enso/blob/a2fc57080e6cd19720aebe4e60befa0634a90ae0/test/Base_Tests/src/Main.enso#L174 so it is included in all the tests. But it is still marked as *pending*

Reply:
Added to Main. I traced the failing line `huge_a == huge_a_int` to `EqualsNode`, where it attems to find a conversion from `Integer` to `Decimal`. It reaches `InvokeConversionNode.doFallback`, which is reporting that the conversion does not exist, and so `==` returns `false`. The conversion does exist in `Decimal.enso`:

```
Decimal.from (that : Integer) =
    IO.println "D.f I"
    Decimal.new that
```

Reply:
Added.